src/Engines/ReduxEngine.js

```diff
@@ -0,0 +1,53 @@
+import { connect } from 'react-redux';
+
+class StateManager {
+  stateChangeHandler = null;
+  reducer = null;
+  reduxDispatch = null;
+
+  constructor(dispatch) {
+    this.reduxDispatch = dispatch;
+  }
+
+  getState(props) {
+    return props.state;
+  }
+
+  setupHandlers = (reducer, setChangedHandler) => {
+    this.reducer = reducer;
+    this.stateChangeHandler = setChangedHandler;
+  }
+
+  dispatch = (cmd) => {
+    this.reduxDispatch(cmd);
+  }
+}
+
+let stateManagerInstance = null;
+
+function createEngine(dispatch) {
+  stateManagerInstance = new StateManager(dispatch);
+
+  return {
+    appHoc: connect(mapStateToProps),
+    stateManager: stateManagerInstance,
+  };
+}
+
+function mapStateToProps(state) {
+  return {
+    state: state.nav,
+  };
+}
+
+export function navReducer(state, cmd) {
+  const newState = stateManagerInstance !== null && typeof stateManagerInstance.reducer === 'function' ? stateManagerInstance.reducer(!state || state.index === undefined ? null : state, cmd) : null;
+
+  if (!newState) return state || {};
+
+  setTimeout(function getOuttaHere() { stateManagerInstance.stateChangeHandler(newState); });
```

Comment:
This might belong in a middleware but I still have to figure that out... If we need a middleware to get `react-native-router-flux` to work with `redux` that is slightly more complexity on the end users hand. On the other side it gives them more control of wiring it into their setup. 

This is wrapped in a `setTimeout` because I needed it to be executed outside of the current context. If it executes while `redux` is doing its reducer thing then it moans that you are `dispatch`ing while its reducing. Wrapping it in the `setTimeout` is a hack no doubt but it allows for the `dispatch`s that occur in `stateChangeHandler` to execute correctly.